### ISSUE TITLE
Add option to geoJson to set viewCoords

### ISSUE THREAD

**Issue by supportdesk-si** [apache/echarts#20451]
### What problem does this feature solve?

Currently eCharts allow to set boundingBox, however zoom and pan are not working outside of the boundingCoords. The goal is to open chart that is always fitting the desired area, yet allowing to move freely around the geoJson map.

### What does the proposed API look like?

```
viewCoords: [
    // [lng, lat] of left-top corner
    [-180, 90],
    // [lng, lat] of right-bottom corner
    [180, -90]
]
```

**Comment by helgasoft** [comment]
related to #20412 where workaround is given

**Comment by helgasoft** [comment]
Note: Would be better to reply instead of editing the original post. Conversation gets confusing.

> the workaround is not a viable solution
**the most important part**, on chart load or data refresh, the focus of the map will be where the data is.

The workaround has already this functionality, Try refresh with _toolbox.feature.restore_ - chart goes back to original center/zoom, [Demo]().
So the question remains "what is the workaround solution missing ?". My POV is that new feature _viewCoords_ is not needed.

**Comment by supportdesk-si** [comment]
Sure, here is the issue. 

1. Make sure the browser window only covers part of the screen.
2. Click inside the chart and hold the left button on the mouse
3. Drag the mouse so it goes outside the browser window, notice that click event is still registered and map continues to move with the mouse
4. Make sure the area in your example is completely off the screen.
5. Release the left button on the mouse
6. Try to zoom or pan on the map

Attaching small animation showing the issue

![firefox_Vp8wRet6NN](https://github.com/user-attachments/assets/7934437e-cfae-44a5-afb3-ad38be71bdc0)

**Comment by helgasoft** [comment]
ok, you are explaining the _boundingBox_ problem. I acknowledge it and proposed a workaround with center/zoom.
Sorry, the Demo link in my previous post is empty. Here is the [Demo](https://echarts.apache.org/examples/en/editor.html?code=PTAEBUAsEsGdQKYA8CGBbADgGwaATggI4Cu0B8AogMKQp4Au8AbgKwB0ALGwAygD2eUFhT0EeAFBYsSUAF5QAWgAc7AGwslAZlWaA3EKwBPOaE0B2NgE5NAJi2rd44nhnzlXc2bOaAjPufGsgCQmpZsPiw2ZhwsjgBGfMQAdgAm0EkA5lR8Aimw8gDaoAVSSAA0BoYAuhUFzuVBAVWgVY5ohjR09GywkHwA7gAyfChpmQAUAJSOACRsGQj04wBKAPKr4AD6AAoAguAAEqAA1KAA5MApIijAKLCwi8ALfMAAqgDKu2wAVrB8SWcKgAzZIAY3o0H-oHGxFgKAAUn8kpNQABvcSgRCg2gMWBsAgZOCiPAAWRQGHGZw-u0BoFhCKRFXRmMxu2EsAA1igAFxojEszE4IH0XkKHy-Mr8gX0PgYXk2FiSgWY_rQFL0SC8iJS0AAXyVLIOKH6KGg0F5zOVQpFih8Pm4BulsvlSkdLNV6s1oBYOv1OrO22IYhloGW0FBfDOFp1goQwtFZlUbsxMrloBsSZjoA9Gvlvv5uum_JBSXBkKS-BQqT4aG20AQ7zE9dg41BCCSxIqeFG0FhKMtmIjSVg9FAV3oKBMBTONIqZwAQrSzlQlwARM5VNhocnjcb0FGyAB8fOV-EWzgrA-VTBQWCDvLJGvxiVS40fkHxVZSNamoAAVKA9rcJMyYskk6AILyy4iAgGQCMYZwnKA9BZrqjgCoW6EsgQ9AXieyr0IYGCQecGD1oCWYRrk6Qwe8hgjggaBQc8FGnjKfBYBCaZXgKQICNu9DElBqJxLqFqgrq0KoikuoAKSTGcqGgcIcQIFg0anpivQDLyQK3g8SlZipamDOkJE8Sy2n9Lp-kIIZp5VtAAnlquzgiOWvIOlm3ZpLCoFth2YigeOKC-lhur8rKEJQvIV7PBpyrbmmVKfKxyp4CMTHIXgQb-e2QnFMo7DcD4SjcIqphKGwLAcFElg1KAABeORZZoXmniACTJGMWQ5HgeS-m67GcdA3H6shORYAkSDcqiQIICIziQai5AygQs26ptbo4AsqQWn6AoPHgza8gUWaoshREkWcWBmbAtIYBxhi3UkkH0DlCAVFmgpme8hE4OJHECNyZwECktI5pqLAHZpIXcgUF1Uf1sDw99LIlNIZRBFI1S1KUWNNF9mno_UZQBEEDV1C4FQ45T-PY0YzTNLqNT2elX41nWDZNggLYFEVPCleVFSaFVNV1Q1EQgd5HO1vWjbHbz4z83aqhsComgsNwHAi2ESgNQq0sClUBZYe0nQMGwMApAgww9hMRaYubOLdA89CrBg0VJOMUXlkWmFAA)

**Comment by helgasoft** [comment]
> relies on **zoom** level being manually set to fit the area into the screen

correct, zoom is **manual** setting, there is no easy way of calculating it from data. 
So I agree that the problem is deeper than originally thought.
And since there is no data-driven initialization workaround. I now think _boundingCoords_ needs to be **fixed**. Limit the roaming to the initial boundary does not make sense. And no other charting library is doing that AFAIK.

**Comment by supportdesk-si** [comment]
Great!!! There are two possible scenarios, either add an option `viewCoords`, or add an option _is navigation restricted_ `isNavRestriced : [true|false]` with default value being `true` The later will be compatible with previous releases and will require no updates from those who use the library already.
